fe/fe-core/src/main/java/org/apache/doris/common/plugin/PluginFileCache.java

```diff
@@ -0,0 +1,103 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.common.plugin;
+
+import com.google.common.base.Strings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * PluginFileCache - Simple static memory cache for plugin file validation.
+ * <p>
+ * Caching logic:
+ * 1. Users do not provide MD5 -> Use it directly if there are files locally (most efficient)
+ * 2. User provides MD5 -> Quick check using cached MD5 (avoid recalculation)
+ */
+public class PluginFileCache {
+    private static final Logger LOG = LogManager.getLogger(PluginFileCache.class);
+
+    // Cache: localPath -> FileInfo
+    private static final ConcurrentHashMap<String, FileInfo> CACHE = new ConcurrentHashMap<>();
```

Comment:
i think you should use CaffineCache to expire the unused entry automatically.
There is no way to clear the cache, currently.
Same issue for BE's cache.BE has lru cache that can do this